src/App/MessagePageClass.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import ErrorPng from './media/error.png';
+import Button from '@material-ui/core/Button';
+import HomeIcon from '@material-ui/icons/Home';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import FreeBoardGameBar from './FreeBoardGameBar';
+import SvgError from './media/SvgError';
+import Typography from '@material-ui/core/Typography';
+import { Link } from 'react-router-dom';
+
+interface IMessageState {
+  linkHidden: boolean;
+  startTime: number;
+}
+
+interface IMessageProps {
+  type: string;
+  message: string;
+}
+
+export class MessagePage extends React.Component<IMessageProps, IMessageState> {
+  constructor(props: IMessageProps) {
+    super(props);
+    this.state = {
+      linkHidden: props.type !== 'error',
+      startTime: Date.now(),
+    };
+    if (props.type !== 'error') {
+      requestAnimationFrame(this._animate(Date.now()));
+    }
+  }
+
+  _animate = (now: number) => () => {
+    const elapsed = now - this.state.startTime;
+    const timedout = elapsed > 5000;
+    const linkHidden = timedout ? false : this.props.type !== 'error';
```

Comment:
I don't think you need to check the type here given that _animate won't be called in case of errors. So in effect link hidden === timed out, you can even remove the link hidden variable.

Reply:
You're right, thank you!